compiler/noirc_frontend/src/hir/comptime/errors.rs

```diff
@@ -1,47 +1,155 @@
+use std::rc::Rc;
+
+use crate::{
+    hir::def_collector::dc_crate::CompilationError, parser::ParserError, token::Tokens, Type,
+};
 use acvm::{acir::AcirField, FieldElement};
+use iter_extended::vecmap;
 use noirc_errors::{CustomDiagnostic, Location};
 
 use super::value::Value;
 
 /// The possible errors that can halt the interpreter.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub enum InterpreterError {
-    ArgumentCountMismatch { expected: usize, actual: usize, location: Location },
-    TypeMismatch { expected: Type, value: Value, location: Location },
-    NonComptimeVarReferenced { name: String, location: Location },
-    IntegerOutOfRangeForType { value: FieldElement, typ: Type, location: Location },
-    ErrorNodeEncountered { location: Location },
-    NonFunctionCalled { value: Value, location: Location },
-    NonBoolUsedInIf { value: Value, location: Location },
-    NonBoolUsedInConstrain { value: Value, location: Location },
-    FailingConstraint { message: Option<Value>, location: Location },
-    NoMethodFound { name: String, typ: Type, location: Location },
-    NonIntegerUsedInLoop { value: Value, location: Location },
-    NonPointerDereferenced { value: Value, location: Location },
-    NonTupleOrStructInMemberAccess { value: Value, location: Location },
-    NonArrayIndexed { value: Value, location: Location },
-    NonIntegerUsedAsIndex { value: Value, location: Location },
-    NonIntegerIntegerLiteral { typ: Type, location: Location },
-    NonIntegerArrayLength { typ: Type, location: Location },
-    NonNumericCasted { value: Value, location: Location },
-    IndexOutOfBounds { index: usize, length: usize, location: Location },
-    ExpectedStructToHaveField { value: Value, field_name: String, location: Location },
-    TypeUnsupported { typ: Type, location: Location },
-    InvalidValueForUnary { value: Value, operator: &'static str, location: Location },
-    InvalidValuesForBinary { lhs: Value, rhs: Value, operator: &'static str, location: Location },
-    CastToNonNumericType { typ: Type, location: Location },
-    QuoteInRuntimeCode { location: Location },
-    NonStructInConstructor { typ: Type, location: Location },
-    CannotInlineMacro { value: Value, location: Location },
-    UnquoteFoundDuringEvaluation { location: Location },
+    ArgumentCountMismatch {
+        expected: usize,
+        actual: usize,
+        location: Location,
+    },
+    TypeMismatch {
+        expected: Type,
+        value: Value,
+        location: Location,
+    },
+    NonComptimeVarReferenced {
+        name: String,
+        location: Location,
+    },
+    IntegerOutOfRangeForType {
+        value: FieldElement,
+        typ: Type,
+        location: Location,
+    },
+    ErrorNodeEncountered {
+        location: Location,
+    },
+    NonFunctionCalled {
+        value: Value,
+        location: Location,
+    },
+    NonBoolUsedInIf {
+        value: Value,
+        location: Location,
+    },
```

Comment:
We didn't change our formatter settings, right? Odd that this file got hit with a reformatting

Reply:
Interesting, removing the new variant I added reverts the formatting changes for all the other variants. So adding one variant can affecting the formatting of all the others...

Reply:
I think I found a formatting bug. I removed the `fm::` prefix from `fm::FileId` and that fixed the formatting for every field. It isn't just a length requirement though, since changing the `message: &'static str` to just `message: String` (net 6 chars saved over the net 4 from removing `fm::`) doesn't fix the formatting.